python/across_api/base/constraints.py

```diff
@@ -41,7 +41,7 @@ def get_slice(time: Time, ephem: EphemBase) -> slice:
     else:
         # Check that the time range is within the ephemeris range, as above.
         assert (
-            time[0] >= ephem.begin and time[-1] <= ephem.end
+            time[0].unix >= ephem.begin.unix and time[-1].unix <= ephem.end.unix
```

Comment:
Is it important to convert to UNIX time? Does astropy.time.Time not support comparison between Time objects with differing scales and representations?

Reply:
Actually something more fundamental may be happening here. Consider this code:

```python
from time import time
from astropy.time import Time
from datetime import datetime

t = time()
dt = datetime.utcfromtimestamp(t)

assert Time(t, format='unix').unix == Time(dt.utcfromtimestamp(t)).unix, "This should work"
assert Time(t, format='unix') == Time(dt.utcfromtimestamp(t)), "This won't work"
```
It's not clear to me why, but comparing `Time` constructed from a `datetime` and a unix time are not equal.

Reply:
Consider this code:

```python
from time import time
from astropy.time import Time
from datetime import datetime

t = time()
dt = datetime.utcfromtimestamp(t)
assert Time(t, format='unix').unix == Time(dt).unix, "This works"
assert Time(t, format='unix') == Time(dt), "This won't work"
```
It fails at the 2nd assertion, despite the two sides of the assertion being generated from the same timestamp.

Reply:
Looks like astropy has an `isclose` method in `Time` to deal with this sort of thing, but it's not obvious to me how you'd do `>=` with it.
```python
assert Time(t, format='unix').isclose(Time(dt), 1e-6 * u.s)
```

Reply:
Subsequent to this line, you call `ephindex` which computes an integer index. What you really want to guard against is that this index is within bounds. If it's out of bounds, will you get an IndexError, or will it silently do something weird? If the former, you may be able to do without this assertion.

Reply:
It will silently do something weird. Best to keep in this assertion, which should never be triggered unless a bug is introduced.

Reply:
In that case, I suggest lowering this assertion down into `ephindex` and asserting that the index is within bounds. That kind of assertion will not be subject to floating point imprecision.

As written, this assertion assumes that all of the times under comparison have the same scale, which may be true right now but is a latent bug.

Reply:
Sure, makes sense to do it there. Fixed in [ff24853](https://github.com/nasa-gcn/gcn.nasa.gov/pull/1975/commits/ff24853f09067b18e3910804aa9d768f502d822d).